### PR TITLE
CMake: update to 3.13

### PR DESCRIPTION
**Pull request by cujomalainey**
Pushed this to see if Zephyr would completely break if this was set. Feel free to ignore if the CI is red

### CODE REVIEW

**Comment by cujomalainey** [comment]
Zephyr build is green, lets use this feature

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,7 +14,7 @@ if(NOT (${CMAKE_VERSION} VERSION_LESS "3.13.0"))
 	# CMake 3.13+ has less restrictive rules for target_link_libraries()
 	# Until we make 3.13 as required minimum version we want to
 	# use old behaviour for compatibility
-	cmake_policy(SET CMP0079 OLD)
+	cmake_policy(SET CMP0079 NEW)
```
Comment (marc-hb) [inline CMakeLists.txt:17]: I understand this policy only superficially yet I'm fairly confident that `cmake_minimum_required` should be bumped to 3.13 and that the version conditionals immediately above and below should then be either removed or simplified.

**Review by marc-hb — CHANGES_REQUESTED** [review]
@nashif would anyone in the Zephyr team have the CMake expertise and time to review this policy change?

@cujomalainey can you provide some rationale? Is this just to remove a warning or is this meant to unblock some actual and future CMakeLists.txt change? Just wondering.

> Zephyr build is green, lets use this feature

Zephyr is probably the most complex CMake "situation" in SOF, on the other hand it requires a very recent CMake. I was also concerned about https://sof-ci.01.org/sof-pr-viewer/#/build/PR4734/build7206055 which tends to rely on a more "static" build environment but it fortunately uses CMake version 3.13.3 right now.

File: CMakeLists.txt (hunk 1 above)
Comment (lgirdwood) [reply CMakeLists.txt:17]: Lets make cmake min version 3.13 (IIRC Zephyr also did this a few weeks ago). @cujomalainey would this work for your linking (or do we also change the policy) ?

Comment (cujomalainey) [reply CMakeLists.txt:17]: Let me try it out and see, I am fine with that solution as well

**Comment by cujomalainey** [comment]
@marc-hb this legacy behaviour prevents linking to the `sof` executable from anywhere other than the root CMake file (where `sof` is declared`. I.e. third party libs when installing their binary must have linking code all the way back in the root rather than up in the audio directory where they live. Without this you get the following error. So the alternate solution would be to copy config code all the way back and pollute the root cmake file.

```
CMake Error at src/audio/CMakeLists.txt:69 (target_link_libraries):
  Attempt to add link library "libhifi3_google_hotword_dsp_api.a" to target
  "sof" which is not built in this directory.

  This is allowed only when policy CMP0079 is set to NEW.


CMake Error at src/audio/CMakeLists.txt:70 (target_link_libraries):
  Attempt to add link library "c" to target "sof" which is not built in this
  directory.

  This is allowed only when policy CMP0079 is set to NEW.


CMake Error at src/audio/CMakeLists.txt:71 (target_link_libraries):
  Attempt to add link library "m" to target "sof" which is not built in this
  directory.

  This is allowed only when policy CMP0079 is set to NEW.


-- Configuring incomplete, errors occurred!
```

File: CMakeLists.txt (hunk 1 above)
Comment (lgirdwood) [reply CMakeLists.txt:17]: Fwiw, I notice Zephyr is at ~3.20.x and we should mirror Zephyr going forwards.

**Comment by marc-hb** [comment]
Thanks @cujomalainey I think this explanation deserves to be in the commit message. This also clarifies that you indeed want that NEW 3.13 feature to be a requirement and that you are not just silencing some warning. Once 3.13 is a requirement then most of the CMake code around this policy can be deleted. Good riddance.

> Fwiw, I notice Zephyr is at ~3.20.x and we should mirror Zephyr going forwards.

I don't think that's necessary for now. Our XTOS build does not require any bleeding edge CMake feature, 3.13 seems plenty enough. 3.20 is very recent and it would break at least quickbuild CI and probably many other places. Most Linux distributions don't have it yet. This 3.20 requirement broke the Zephyr build for many.

The CMake project is incredibly careful about backward compatibility so having 3.13 code in SOF and 3.20 code in Zephyr should not be a problem.

**Comment by lgirdwood** [comment]
> > Fwiw, I notice Zephyr is at ~3.20.x and we should mirror Zephyr going forwards.
> 
> I don't think that's necessary for now. Our XTOS build does not require any bleeding edge CMake feature, 3.13 seems plenty enough. 3.20 is very recent and it would break at least quickbuild CI and probably many other places. Most Linux distributions don't have it yet. This 3.20 requirement broke the Zephyr build for many.
> 
> The CMake project is incredibly careful about backward compatibility so having 3.13 code in SOF and 3.20 code in Zephyr should not be a problem.

ok, so for now we are probably fine, but after v1.9 and if 21.10 distros and Windows move to 3.20 (or newer than current min) we should do like wise and align or close the gap with Zephyr. No point in carrying baggage for older cmake versions.

**Comment by marc-hb** [comment]
> No point in carrying baggage for older cmake versions.

Totally agree we should not hold back and bump the requirement as soon as we need some specific CMake feature or bug fix - exactly like here. But there is a trade-off: being ahead of even the latest Linux distros is painful as all Zephyr users recently discovered. BTW Zephyr made this bleeding-edge 3.20 bump reluctantly because there was some specific CMake feature or bug fix required by its very complex toolchain abstraction layer; not just because it was the latest and greatest CMake version. I think it used to be 3.13 before that?

**Review by marc-hb — APPROVED** [review]
The 
```
Exception TimeoutError occurred at 'C:\\work\\tools\\pyexecnetcache\\cavs_scripts-py\\utilities\\fw_loader_pci.py':499
                   Cleaning ROM control IPC failed. Platform restart may be required.
```
in https://sof-ci.01.org/sof-pr-viewer/#/build/PR4734/build7215505 is platform specific and seems unrelated. It's the same as https://sof-ci.01.org/sof-pr-viewer/#/build/PR4740/build7214514 and probably others. The build phase was successful.

cc @kkarask 

The APL_UP2_NOCODEC timeouts in https://sof-ci.01.org/sofpr/PR4734/build10273/devicetest/ don't seem related either. It looks like the device was unreachable for some time for some unknown reason.
EDIT: I checked `journalctl`. The system crashed for no obvious reason (and rebooted). The logs just stop suddenly.

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
Unrelated test script timeout on APL, other APL DUTs pass. Unrelated.

**Pull request merged**